library/src/main/kotlin/display/bridgeSearch/implemetation/GraphBridgeSearchTarjan.kt

```diff
@@ -0,0 +1,58 @@
+package display.bridgeSearch.implementation
+
+import display.bridgeSearch.GraphBridgeSearch
+import graph.Graph
+
+class GraphTarjanBridgeFinder<V, E> {
+    private var time = 0
+
+    override fun <V, E> getBridges(graph: Graph<V, E>): List<Pair<V, V>> {
+        val vertices = graph.vertexSet()
+        val disc = mutableMapOf<V, Int>()
+        val low = mutableMapOf<V, Int>()
+        val parent = mutableMapOf<V, V?>()
+        val bridges = mutableListOf<Pair<V, V>>()
+
+        for (v in vertices) {
+            disc[v] = -1
+            low[v] = -1
+            parent[v] = null
+        }
+
+        for (v in vertices) {
+            if (disc[v] == -1) {
+                dfs(graph, v, disc, low, parent, bridges)
+            }
+        }
+
+        return bridges
+    }
+
+    private fun dfs(
+        graph: Graph<V, E>,
+        u: V,
+        disc: MutableMap<V, Int>,
+        low: MutableMap<V, Int>,
+        parent: MutableMap<V, V?>,
+        bridges: MutableList<Pair<V, V>>
+    ) {
+        disc[u] = time
+        low[u] = time
+        time++
+
+        for (e in graph.outgoingEdgesOf(u)) {
+            val v = graph.getEdgeHead(e)
```

Comment:
my bad, нужно было написать где-нибудь, что у нас ребра одинаковые в графе могут быть. такой подход вернет не все ребра, а только их Set. можно написать так

```suggestion
        for (v in graph.outgoingVerteciesOf(u)) {
```